tests/rptest/services/redpanda.py

```diff
@@ -3694,12 +3694,17 @@ def _get_object_storage_report(self,
 
         bucket = self.si_settings.cloud_storage_bucket
         environment = ' '.join(f'{k}=\"{v}\"' for k, v in vars.items())
+        bucket_arity = len(list(self.get_objects_from_si()))
```

Comment:
I'd use `sum(1 for _ self.get_objects_from_si())`. This avoids building a potentially very big list.

Reply:
good idea